### PR TITLE
Pull request for Milestone 3 - Tic Tac Toe

### PR DESCRIPTION
**Pull request by bruna-genz**
In this milestone I created the file game_logic.rb in the bin repository. This file contains the classes and methods used to make the game work. Also, I edited the main.rb file to create a better interaction with the user.

### CODE REVIEW

**Comment by Torres-ssf** [comment]
### Pull Request Under Review By Sérgio (TSE)

File: lib/game_logic.rb [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    end
+  end
+
+  def occupied?(user_input)
+    index = user_input - 1
+    moves_player1.include?(index) || moves_player2.include?(index)
+  end
+
+  def draw?
+    turn >= 9
```
Comment (Torres-ssf) [inline lib/game_logic.rb]: **[REQUIRED]**
We have a bug related to win and draw message. If a player won the match in the last possible move, both draw and winner messages appear:
Screenshot from 2019-10-02 08-14-59

File: lib/game_logic.rb [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    "#{board[6]}|#{board[7]}|#{board[8]}\n"
+  end
+
+  def mark_position(index, mark)
+    board[index] = ' ' + mark + ' '
+  end
+end
+
+class Player
+  attr_accessor :name, :mark
```
Comment (Torres-ssf) [inline lib/game_logic.rb]: **[REQUIRED]**
 The player's `name` never changes after been assigned. You can safely use `attr_reader` for it.

File: lib/game_logic.rb [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    else
+      result = 'Please, type yes or no.'
+    end
+    result
+  end
+end
+
+class Board
+  attr_accessor :board
+  attr_reader :mark
```
Comment (Torres-ssf) [inline lib/game_logic.rb]: **[REQUIRED]**
After been assigned, the board's value never changes. We can change the encapsulation here making `board` attr_reader.

**Comment by bruna-genz** [comment]
In the last commit, I fixed the bug from when a player won the match in the last possible move and changed the variables board and name to attr_reader.

**Pull request merged**